src/test/java/org/folio/rest/api/InventoryHierarchyViewTest.java

```diff
@@ -98,24 +97,22 @@ public void setUp() {
   }
 
   @Test
-  public void ServerErrorWrittenOutOnDatabaseError() throws InterruptedException, ExecutionException, TimeoutException {
-
-    setFaultyStatisticalCode();
+  public void serverErrorWrittenOutOnDatabaseError() throws InterruptedException, ExecutionException, TimeoutException {
 
+    setFaultyViewFunction();
     params.put(QUERY_PARAM_NAME_SKIP_SUPPRESSED_FROM_DISCOVERY_RECORDS, "false");
 
     List<JsonObject> instances = requestInventoryHierarchyViewUpdatedInstanceIds(params, response -> assertThat(response.getStatusCode(), is(200)));
     UUID[] instanceIds = instances.stream()
-      .map(json -> UUID.fromString(json.getString("instanceId")))
-      .toArray(UUID[]::new);
-
-    List<JsonObject> instancesData = requestInventoryHierarchyItemsAndHoldingsViewInstance(
-      instanceIds, false, response -> assertThat(response.getStatusCode(), is(500)));
-
-    JsonObject error = instancesData.get(0);
-    assertThat(error.getString("message"), is("invalid input syntax for type uuid: \"5t632 ytbg vnc\""));
-    removeFaultyStatisticalCode();
-
+        .map(json -> UUID.fromString(json.getString("instanceId")))
+        .toArray(UUID[]::new);
+
+    requestInventoryHierarchyItemsAndHoldingsViewInstance(instanceIds, false, response -> {
+      undoFaultyViewFunction();
+      assertThat(response.getStatusCode(), is(500));
+      String message = new JsonObject(response.getBody()).getString("message");
+      assertThat(message, is("function get_items_and_holdings_view(unknown, unknown) does not exist"));
```

Comment:
Why did these changes result in the behaviour of the hierarchy view (which is purely a read) changing?

Reply:
Because the test setup of serverErrorWrittenOutOnDatabaseError() changed from setFaultyStatisticalCode() to setFaultyViewFunction() - for obvious reasons.